bids-validator/src/schema/applyRules.ts

```diff
@@ -442,6 +444,43 @@ function evalJsonCheck(
         ]);
       }
     }
+    let originFileKey = ''
+    if (keyName in context.sidecarKeyOrigin) {
+      originFileKey = `${context.sidecarKeyOrigin[keyName]}.${keyName}`
```

Comment:
Since `.` is a part of the path, might be a little clearner to use a different separator:

```suggestion
      originFileKey = `${context.sidecarKeyOrigin[keyName]}:${keyName}`
```